logging/logger_registry_test.go

```diff
@@ -13,15 +14,50 @@ var (
 	l3 = NewLogger("logger-3")
 )
 
-func mockRegistry() *loggerRegistry {
-	manager := newLoggerRegistry()
+func mockRegistry() *Registry {
+	manager := newRegistry()
 	manager.registerLogger("logger-1", l1)
 	manager.registerLogger("logger-2", l2)
 	l3.Sublogger("sublogger-1")
 	globalLoggerRegistry = manager
 	return manager
 }
 
+type memAppender struct {
+	entries []*zapcore.Entry
+}
+
+func (mem *memAppender) Write(entry zapcore.Entry, field []zapcore.Field) error {
+	mem.entries = append(mem.entries, &entry)
+	return nil
+}
+
+func (mem *memAppender) Sync() error {
+	return nil
+}
+
+func TestRegistryIsolation(t *testing.T) {
```

Comment:
I started this to show my new code fixed a bug. But then I realized I couldn't even express the bug with the `GetOrNewLogger` free-floating function. So it was pointless. Feel free to delete.